### DISCUSSION TITLE
How to Inherit from DataGrid?

### DISCUSSION THREAD

**Discussion by alvahtin** [AvaloniaUI/Avalonia#15247] [Q&A] [answered]
I want create my own class, inherited from DataGrid. But when I put it in Window, it is unvisible. I think this is because the templates for the DataGrid are not valid for my class. The templates are set to Application.Styles, the DataGrid is visible, but my class is not.

How to inherit templates from DataGrid to my class?

**Comment by workgroupengineering** [comment, marked as answer]
In your custom DataGrid you should override `StyleKeyOverride`

```c#
protected override Type StyleKeyOverride => typeof(DataGrid);
```

**Reply by alvahtin** [reply]
Thank you, very much!
